frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { match } from 'react-router';
+import { k8sGet } from '@console/internal/module/k8s';
+import { ClusterServiceVersionModel } from '@console/operator-lifecycle-manager';
+import { BreadCrumbs } from '@console/internal/components/utils';
+import { Radio, Title } from '@patternfly/react-core';
+import { checkForIndependentSupport } from '../independent-mode/utils';
+import { OCSServiceModel } from '../../models';
+import InstallExternalCluster from '../independent-mode/install';
+import { CreateOCSServiceForm } from './create-form';
+import './install-page.scss';
+
+enum MODES {
+  CONVERGED = 'Converged',
+  INDEPENDENT = 'Independent',
+}
+
+// eslint-disable-next-line no-shadow
+const InstallCluster: React.FC<InstallClusterProps> = ({ match }) => {
+  const {
+    params: { ns, appName },
+    url,
+  } = match;
+
+  const [isIndependent, setIsIndependent] = React.useState(false);
+  const [mode, setMode] = React.useState(MODES.CONVERGED);
+  const [sample, setSample] = React.useState(null);
+  const [clusterServiceVersion, setClusterServiceVersion] = React.useState(null);
+
+  const handleModeChange = (_checked: boolean, event: React.FormEvent<HTMLInputElement>) => {
+    const { value } = event.currentTarget;
+    setMode(value as MODES);
+  };
+
+  React.useEffect(() => {
+    k8sGet(ClusterServiceVersionModel, appName, ns)
+      .then((clusterServiceVersionObj) => {
+        setIsIndependent(checkForIndependentSupport(clusterServiceVersionObj));
+        try {
+          setSample(
+            JSON.parse(_.get(clusterServiceVersionObj.metadata.annotations, 'alm-examples'))[0],
```

Comment:
Optional Chaining instead of `_.get`

Reply:
I have used optional chaining everywhere except here. This is some legacy code from the original flow for installation, which I don't want to touch for now.

Reply:
ack